src/master/matoclserv.cc

```diff
@@ -823,34 +819,21 @@ int matoclserv_load_sessions() {
 }
 #undef MFSSIGNATURE
 
-int matoclserv_insert_openfile(session* cr,uint32_t inode) {
-	filelist *ofptr,**ofpptr;
-	int status;
-
-	ofpptr = &(cr->openedfiles);
-	while ((ofptr=*ofpptr)) {
-		if (ofptr->inode==inode) {
-			return SAUNAFS_STATUS_OK;       // file already acquired - nothing to do
-		}
-		if (ofptr->inode>inode) {
-			break;
-		}
-		ofpptr = &(ofptr->next);
+int matoclserv_insert_openfile(session *cr, uint32_t inode) {
+	if (cr->openedfiles.count(inode) > 0) {
```

Comment:
Consider using contains instead of count.

Reply:
Applied.